File: TestNodeJs/models/asset.model.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Asset extends Model {
    static associate(models) {
      this.belongsToMany(models.Person, {
        through: "Person_Assets",
        foreignKey: "asset_id",
        as: "people",
        timestamps: false,
      });

      this.belongsToMany(models.Family, {
        through: "Family_Assets",
        foreignKey: "asset_id",
        as: "families",
        timestamps: false,
      });
    }
  }
  Asset.init(
    {
      id: {
        type: DataTypes.INTEGER.UNSIGNED,
        allowNull: false,
        primaryKey: true,
        autoIncrement: true,
      },
      asset: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      harga: {
        type: DataTypes.INTEGER,
        allowNull: false,
      },
    },
    {
      sequelize,
      paranoid: true,
      modelName: "Asset",
      tableName: "Assets",
      createdAt: "created_at",
      updatedAt: "updated_at",
      deletedAt: "deleted_at",
    }
  );
  return Asset;
};

File: TestGolang/models/asset.model.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Asset struct {
	Id        uint           `gorm:"primaryKey,autoIncrement" json:"id"`
	Asset     string         `gorm:"not null" json:"asset"`
	Harga     int            `gorm:"not null" json:"harga"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"null" json:"deleted_at"`
}

type FamilyAsset struct {
	Family_id uint
	Asset_id  uint
}

type PersonAsset struct {
	Person_id uint
	Asset_id  uint
}

func (Asset) TableName() string {
	return "assets"
}
